service-impl/cache-service-impl/simple-cache-service/src/main/java/uk/gov/gchq/palisade/cache/service/impl/SimpleCacheObject.java

```diff
@@ -40,17 +39,34 @@
      */
     private final Optional<byte[]> value;
 
+    /**
+     * Extra information about this entry and it's retrieval.
+     */
+    private final Optional<CacheMetadata> metadata;
+
     /**
      * Create a cache object.
      *
      * @param valueClass the type of the value being cached
      * @param value      the optional cache value, may be empty if no valid entry is present
+     * @param metadata   information about the cache entry
      */
-    public SimpleCacheObject(final Class<?> valueClass, final Optional<byte[]> value) {
+    public SimpleCacheObject(final Class<?> valueClass, final Optional<byte[]> value, final Optional<CacheMetadata> metadata) {
         Objects.requireNonNull(valueClass, "valueClass");
         Objects.requireNonNull(value, "value");
+        Objects.requireNonNull(metadata, "metadata");
         this.valueClass = valueClass;
         this.value = value;
+        this.metadata = metadata;
+    }
+
+    /**
+     * Get the metadata about this cache entry.
+     *
+     * @return the metadata
+     */
+    public Optional<CacheMetadata> metadata() {
```

Comment:
Create a set metadata and set value, or a extract metadata method. Depends on where we want the metadata encoding/decoding logic.

Reply:
Done